docs/README.md

```diff
@@ -0,0 +1,77 @@
+# Codeshare
+
+CodeShare is a web-based real-time code sharing tool. It empowers developers to collaborate remotely during hackathons, contests or for a project. It can also be potentially used in mentorship programs to live share code. The tool also compiles and runs code.
+
+## Team members:
+1. Rohit M P
+2. Shanthanu S Rai
+3. Ameya N Deshpande
+4. Harshvardhan R
+5. Vaibhav Puri
+
+## Objectives:
+
+Following were the objectives:
+1. Compare Codemiror vs Monaco and integrate either of them.
+2. To create a server-based system using Websockets (Channel).
+3. Display the users and their respective cursors on the editor.
+4. Implement CRDT functions to handle real time problems 
+5. Create different sessions (Rooms) on the Channel.
+6. Maintain a database for recording all the changes in the editor.
+7. Handle newly joined user and last user.
+8. Provide code compilation and execution for different languages.
+
+## Methodology
+
+Codeshare is a "real-time" code sharing tool i.e. all the changes made in one editor are applied immediately to other editors on the same session.The project is based on a Central Relay Server which manages the connection between all the users of a session and maintains a database for it.
+
+While trying to achieve real-time collaboration, we do not make use of blocking statements or try to serialize the opertions by different users (For example, insert character, delete charater, insert newline, etc.). We try and handle all the possible operations in such a way that the following properties hold:
+1. **Commutativity**: Concurrent Insert and Delete operations must converge to the same result regardless of the order in which they are applied.
+2. **Idempotency**: Repeated delete operations produce the same result.
+
+Here, is an example why we must take care of these properties:
+![Simultaneous insertion and deletion produce different results](./src/images/non_commutative.png)
+*Source: [https://conclave-team.github.io/conclave-site/](https://conclave-team.github.io/conclave-site/)*
+
+![Duplicate deletion operations are not idempotent](./src/images/non-idempotent.png)
+*Source: [https://conclave-team.github.io/conclave-site/](https://conclave-team.github.io/conclave-site/)*
+
+To take care of the above two properties, we decided on implementing a different structure for our editor, namely **Conflict-Free Replicated Data Type (CRDT)**. Following points explain the essence of CRDT:
```

Comment:
"we decided on maintaining a data structure on each client's side"

Reply:
Done